pallets/offchain-worker/src/lib.rs

```diff
@@ -329,6 +340,21 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		// 
+		fn do_finalize(block_number: T::BlockNumber) {
+			let query_session_length: usize = T::QuerySessionLength::get() as usize;
+			let index_in_session = TryInto::<usize>::try_into(block_number).map_or(query_session_length, |bn| bn % query_session_length);
+			let last_block_number = query_session_length - 1;
+
+			// Clear claim at the first block of a session
+			if index_in_session == 0 {
+				Self::clear_claim();
+			// Do aggregation at last block of a session
+			} else if index_in_session == last_block_number {
+				Self::aggregate_query_result();
```

Comment:
What does this mean? All query is sent at the end of a session?

Reply:
query result can be submitted anytime, we just do aggregation at the end of a session.